### PR TITLE
INS-2639: add kong 3.x support for inso generate k8s config

### PR DESCRIPTION
**Pull request by filfreire**
changelog(Improvements): Added minor Kong 3.x support for generating k8s configs using Inso CLI

While ApiOps v2 is not in full GA/production yet, adding this small QoL improvement to config generation to unblock folks still using Inso CLI for kong on k8s config generation cc @mheap @rspurgeon

### CODE REVIEW

File: packages/openapi-2-kong/src/kubernetes/generate.test.ts [hunk 1]
```diff
@@ -304,6 +304,11 @@ describe('index', () => {
       expect(generateServicePath(serverBasePath)).toBe('/api/.*/.*');
     });
 
+    it('adds /~ for kong 3.+ non legacy configs', () => {
+      const serverBasePath = '/api/.*';
+      expect(generateServicePath(serverBasePath, '', false)).toBe('/~/api/.*/.*');
```
Comment (mheap) [inline packages/openapi-2-kong/src/kubernetes/generate.test.ts]:
```suggestion
      expect(generateServicePath(serverBasePath, '', false)).toBe('/~/api/.*');
```

Comment (filfreire) [reply packages/openapi-2-kong/src/kubernetes/generate.test.ts]: @mheap should we also edit then the `const serverBasePath = '/api/.*';` above to be just `/api` or the wildcard should just not be added if it's already on the path?

Comment (mheap) [reply packages/openapi-2-kong/src/kubernetes/generate.test.ts]: Ah, `.*` isn't supported in the base path

**Review by mheap — APPROVED** [review]

**Review by jackkav — APPROVED** [review]
Still don't agree with legacy defaulting to true, but since this feature is moving to a Go implementation, I'm happy to approve one liners.

**Pull request merged**